server/mqtt.go

```diff
@@ -2138,20 +2138,20 @@ func (as *mqttAccountSessionManager) handleRetainedMsg(key string, rm *mqttRetai
 
 			// Update the in-memory retained message cache but only for messages
 			// that are already in the cache, i.e. have been (recently) used.
-			if erm.loaded != nil {
-				erm.loaded = rm.loaded
-				erm.loaded.expiresFromCache = time.Now().Add(mqttRetainedCacheTTL)
+			if rm != nil {
+				if _, ok := as.rmsCache.Load(key); ok {
+					toStore := *rm
+					toStore.expiresFromCache = time.Now().Add(mqttRetainedCacheTTL)
+					as.rmsCache.Store(rm.Subject, toStore)
```

Comment:
Why not use `key` here?

Reply:
I understand that we need to make a copy and store again, but should we store a `*mqttRetainedMsg` instead?